File: daily/Arrays/1935.Maximum-number-of-words-you-can-type.py
from typing import Set


class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        broken: Set[str] = set(brokenLetters)
        count = 0

        for word in text.split():
            if all(ch not in broken for ch in word):
                count += 1

        return count

File: daily/Arrays/1935.Maximum-number-of-words-you-can-type.go
package daily

import "strings"

func canBeTypedWords(text string, brokenLetters string) int {
	brokenMap := make(map[rune]bool)
	for _, c := range brokenLetters {
		brokenMap[rune(c)] = true
	}

	res := 0
	words := strings.Split(text, " ")
	for _, word := range words {
		for _, c := range word {
			if brokenMap[rune(c)] {
				res++
				break
			}
		}
	}

	return len(words) - res
}
